### PR TITLE
Katherine's Website

### PR DESCRIPTION
**Pull request by ng-david**

### CODE REVIEW

File: css/main.css [hunk 1]
```diff
@@ -0,0 +1,230 @@
+/* BASE STYLES */
+
+body {
+    font-family: 'montserrat';
```
Comment (ng-david) [inline css/main.css:4]: Over here, it's best to define a list of fonts, this way if someone visiting your website does not have `montserrat` for whatever reason (i.e., google font servers are down), they can still read your website with similar fonts. 

https://stackoverflow.com/questions/32348867/what-is-the-significance-of-giving-multiple-font-family-name

File: css/main.css [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+    color: #000000;
+}
+
+/* GRID */
+
+.full-width {
+    width: 1200px;
```
Comment (ng-david) [inline css/main.css:11]: I would suggest reading through the internet about the significance of `rem` units over `px` units! Not a requirement for sure, especially not for personal projects, but in the real world these are probably much more important, so it's good to have context on what they are.

https://snook.ca/archives/html_and_css/font-size-with-rem
https://www.google.com/search?q=rem+vs+px&oq=rem+vs+px&aqs=chrome.0.69i59j69i57j0i271l3j69i60l3.2002j0j7&sourceid=chrome&ie=UTF-8

File: css/main.css [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+    margin: 0 auto;
+    overflow: auto;
+}
+
+.half-width {
+    width: 600px;
+    float: left;
```
Comment (ng-david) [inline css/main.css:18]: Floating is a really great (and tricky) thing to know! But even better than float is that nowadays in CSS3 there is a complex thing called "flexbox." Flexbox is 200% more powerful and a powerful strategy to know about

https://css-tricks.com/snippets/css/a-guide-to-flexbox/

File: css/main.css [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+    text-align: center;
+    font-size: 55px;
+    font-family: lora;
+    padding: 300px 0 0 350px;
+}
+
+.cover h2 {
+    font-size: 36px;
+    text-align: center;
+    font-family: 'montserrat';
```
Comment (ng-david) [inline css/main.css:70]: _nit_ Did you need to redefine this font for every heading type? Or would defining it in the `body { }` be good enough?

File: css/main.css [hunk 5]
```diff
@@ -0,0 +1,230 @@
+/* BASE STYLES */
```
Comment (ng-david) [inline css/main.css:1]: _nit_ I recommend changing the indent settings on your code editor to be 2 spaces instead of 4 spaces. This helps make complex projects more readable because you can fit more on your screen at a time :)

File: css/main.css [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+nav {
+    float: right;
+    padding: 55px 25px 0 25px;
+}
+
+nav ul li {
+    display: inline-block;
+}
+
+nav ul li a {
```
Comment (ng-david) [inline css/main.css:44]: Excellent cascading of CSS selectors!

File: css/main.css [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+    font-size: 36px;
+    text-align: center;
+    font-family: 'montserrat';
+    font-weight: normal;
+    padding: 0 200px 0 450px;
+}
+
+/* Portfolio Section */
+#portfolio {
+    height: 1150px;
```
Comment (ng-david) [inline css/main.css:77]: Imo it's best to avoid defining heights on these "sections". 

This is because we cannot predict the height of these sections. Someday we might want to add more elements to each section (ex: You want to add 3 new projects to your website), but when we do that they're explode out of this section! Fortunately, by just removing this height declaration on line 77, our website becomes much more fluid and would grow the `portfolio` section automatically with each new portfolio added!

TL;DR you can remove this and your website will look pretty much the same as it does now, while making your future easier

File: css/main.css [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+    background-size: cover;
```
Comment (ng-david) [inline css/main.css:78]: _nit_ background-size cover is doing nothing here I believe

File: css/main.css [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+}
+
+.project {
+    background-color: #ebb9ad;
+    height: 80px;
```
Comment (ng-david) [inline css/main.css:83]: _nit_ This height is unnecessary :)

File: css/main.css [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+    background-size: cover;
```
Comment (ng-david) [inline css/main.css:84]: _nit_ No need to define background-size here

File: css/main.css [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+    margin: 0 auto;
```
Comment (ng-david) [inline css/main.css:85]: Margin 0 auto is the most important css line you'll ever learn 👏 😂

File: css/main.css [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+}
+
+.project h3 {
```
Comment (ng-david) [inline css/main.css:88]: It's great you're isolating your h3 styles, but something you may notice is that some of your h3 elements share the same styles: font-size, text-align, font-family, color, padding.

You can extract these to a higher level so there's less code duplication and easier maintenance

```
h3 {
 ...
}

.project h3 {
 ...
 // things specific to project's h3
}
```

File: css/main.css [hunk 13]
```diff
@@ -0,0 +1,230 @@
...
+    font-size: 36px;
+    text-align: center;
+    font-family: lora;
+    color: white;
+    margin: 0 auto;
```
Comment (ng-david) [inline css/main.css:93]: _nit_ I think you might not need margin 0 auto here, might not*

File: css/main.css [hunk 14]
```diff
@@ -0,0 +1,230 @@
...
+    padding: 17px 0 17px 0;
+}
+
+.box {
+    height: 350px;
```
Comment (ng-david) [inline css/main.css:98]: Maintaining these heights as time goes on will definitely get hard if this website was larger. In those cases, defining your project's as individual project rows over the two columns for imgs/descriptions might prove to be easier.

In general, we want to try to define our website in rows so that responsive behavior (shrinking a website down for mobile phones) works easier. Flexbox is a very powerful way to accomplish this, albeit complex.

Bootstrap4 is a much easier entry way to understanding this, but I think you need not focus on this until later in your learning.

Just an FYI for now!

File: css/main.css [hunk 15]
```diff
@@ -0,0 +1,230 @@
...
+    width: 600px;
+    float: left;
+}
+
+.screenshot {
+    height: 300px;
+    Width: 490px;
```
Comment (ng-david) [inline css/main.css:105]: _nit_ I don't believe Width should be capitalized here

File: css/main.css [hunk 16]
```diff
@@ -0,0 +1,230 @@
...
+    padding: 15px 50px 15px 50px;
```
Comment (ng-david) [inline css/main.css:106]: _nit_ We can shorten to `15px 50px`, though I know you already know this

File: css/main.css [hunk 17]
```diff
@@ -0,0 +1,230 @@
...
+}
+
+.proj-title h4 {
+    font-size: 28px;
+    font-family: lora;
+    margin: 50px 0 0 0;
```
Comment (ng-david) [inline css/main.css:112]: _nit_ this can be `margin-top: 50px` instead i believe

File: css/main.css [hunk 18]
```diff
@@ -0,0 +1,230 @@
...
+}
+
+#portfolio p {
+    padding: 10px 0 10px 0;
+    margin: 0;
+}
+
+.description {
+    text-align: justify;
```
Comment (ng-david) [inline css/main.css:121]: Here's a short goofy read about why you might want to use justify 
https://designforhackers.com/blog/justify-text-html-css/#:~:text=Justifying%20text%20disrupts%20that%20even,the%20web%2C%20but%20whenever%20possible.

File: css/main.css [hunk 19]
```diff
@@ -0,0 +1,230 @@
...
+    padding-right: 50px;
+}
+
+/* ABOUT ME */
```
Comment (ng-david) [inline css/main.css:125]: Comments are nice sometimes, but to make maintainability easier and to decrease on unnecessary lines of code, you might notice that you already have prepended each of the following selectors with `#about`, so these comments are actually not too necessary!

File: css/main.css [hunk 20]
```diff
@@ -0,0 +1,230 @@
...
+    font-family: montserrat;
+    font-weight: normal;
+    color: white;
+    padding: 10px 60px 10px 60px;
+    text-align: justify;
+}
+
+#about h4 {
+    font-size: 24px;
+    font-family: montserrat;
```
Comment (ng-david) [inline css/main.css:158]: _nit_ I don't think we need to redefine more times than just in `body {}`.

File: css/main.css [hunk 21]
```diff
@@ -0,0 +1,230 @@
...
+    background-color: #5c7b99;
+}
+
+#contact h3 {
+    font-size: 36px;
+    font-family: lora;
+    color: white;
+    text-align: center;
+    margin: 0 0;
+    padding: 25px 0 25px 0;
```
Comment (ng-david) [inline css/main.css:191]: _nit_ `padding: 25px 0;` 

alternatively

`padding: 25px 0px;` if youre ocd

File: css/main.css [hunk 22]
```diff
@@ -0,0 +1,230 @@
...
+
+#contact h5 {
+    font-size: 20px;
+    font-weight: normal;
+    color: white;
+    text-align: center;
+}
+
+#contact p {
+    font-size: 18px;
```
Comment (ng-david) [inline css/main.css:202]: Let's try to define font-sizes per each element `p` `a` `h1-h5` on the top level (not nested inside these sections) so they can be consistent throughout the website without having to remember which size for which element! :)

File: css/main.css [hunk 23]
```diff
@@ -0,0 +1,230 @@
...
+    font-weight: normal;
+    color: white;
+    text-align: center;
+}
+
+#contact a {
```
Comment (ng-david) [inline css/main.css:208]: If you want to make your links have effects on hover and click and what not, this is a great resource for that:

https://www.w3schools.com/css/css_link.asp

For good UX, it's ideal if your links:
1) Have a different, distinguishing style/color from normal text on the website (though this isn't super necessary when what you have there is an email, more important for other types of links/clickable things)
2) React to the users cursor (outside of the boring hand pointer cursor)

So you can just have it change color on a:hover or something!

File: index.html [hunk 24]
```diff
@@ -0,0 +1,140 @@
+<!doctype html>
+<html lang="en">
+    <head>
+        <title>Katherine Kocen | Protfolio</title>
+<!-- Links to CSS Style Sheets and Fonts -->
```
Comment (ng-david) [inline index.html:5]: _nit_ If you remove this comment, it'll increase professionalism!

File: index.html [hunk 25]
```diff
@@ -0,0 +1,140 @@
...
+            <link rel="stylesheet" href="css/main.css">
+            <link rel="stylesheet" href="css/normalize.css">
```
Comment (ng-david) [inline index.html:7]: Normalize is great. So glad you all get to learn about it!

File: index.html [hunk 26]
```diff
@@ -0,0 +1,140 @@
...
+                            <li><a href="#about">About</a></li>
+                            <li><a href="#contact">Contact</a></li>
+                        </ul>
+                    </nav>
+                </div>    
+            </div>
+
+            <div class="cover">
+                <div class="full-width">
+                    <h1>Hi There! I'm Katherine Kocen.</h1>
```
Comment (ng-david) [inline index.html:30]: _nit_ I think having `There` be capitalized too is a little strange given this section has sentence casing bc of its punctuation 🙈 
https://blog.prototypr.io/sentence-case-or-title-case-5bd581f05d88?gi=43d26eb86d0

File: index.html [hunk 27]
```diff
@@ -0,0 +1,140 @@
...
+                    <h2>Front-end developer, designer, and problem solver!</h2>
+
```
Comment (ng-david) [inline index.html:32]: _nit_ Remove these new lines between each section unless your code blocks become hugeee to increase professionalism if someone looks at this code someday (anyone who visits your website can download it to see how to typed it up)

File: index.html [hunk 28]
```diff
@@ -0,0 +1,140 @@
...
+                    <div class="half-width">
+                        <div class="box">
+                            <img class="screenshot" src="img/rogue-pickings.png" alt="rogue pickings project">
+                        </div>
+
+                        <div class="box">
+                            <img class="screenshot" src="img/recipe-collection.png" alt="recipe collection project">
+                        </div>
+
+                        <div class="box">
```
Comment (ng-david) [inline index.html:53]: Great use of `alt` 👏

**Review by ng-david — COMMENTED** [review]
Looks great!
